build_tools/docker/manage_images.py

```diff
@@ -221,51 +222,67 @@ def update_references(image_url: str, digest: str, dry_run: bool = False):
   # Update references in all grepped files.
   files = completed_process.stdout.split()
   print(f'Updating references in {len(files)} files: {files}')
-  for line in fileinput.input(files=files, inplace=(not dry_run)):
-    print(re.sub(f'{image_url}@{DIGEST_REGEX}', f'{image_url}@{digest}', line),
-          end='')
+  if not dry_run:
+    for line in fileinput.input(files=files, inplace=True):
+      print(re.sub(f'{image_url}@{DIGEST_REGEX}', f'{image_url}@{digest}',
+                   line),
+            end='')
+
+
+# TODO typing
+def get_prod_digests() -> Dict[str, str]:
+  with open(utils.PROD_DIGESTS_PATH, "r") as f:
+    images_with_digests = {}
+    for line in f:
+      url, digest = line.strip().split("@")
+      images_with_digests[url] = digest
+  return images_with_digests
```

Comment:
Ugh, not sure how my suggestion got deleted from the review I sent :|

```suggestion
def parse_prod_digests() -> Dict[str, str]:
  image_urls_to_prod_digests = {}
  with open(utils.PROD_DIGESTS_PATH, "r") as f:
    for line in f:
      image_url, digest = line.strip().split("@")
      image_urls_to_prod_digests[image_url] = digest
  return image_urls_to_prod_digests
```